File: 0001-two-sum/0001-two-sum.java

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int [] arr=new int [nums.length];
        for(int i=0;i<nums.length;i++)arr[i]=nums[i];
        Arrays.sort(arr);
        int l=0,r=nums.length-1;
        while(arr[l]+arr[r]!=target){
            if(arr[l]+arr[r]>target)r--;
            else l++;
        }
        int ans[]= new int[2];
        boolean foundl=false,foundr=false;
      for(int i=0;i<nums.length;i++){
            System.out.println(arr[l]+"\n"+arr[r]);
            if(arr[l]==nums[i]&&!foundl){ans[0]=i;foundl=true;}
            else if(arr[r]==nums[i]&&!foundr){ans[1]=i;foundr=true;}
        } 
        return ans;

    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int>arr(nums.size());
        for(int i=0;i<nums.size();i++){
            arr[i]=nums[i];
        }
        int l=0,r=nums.size()-1;
          sort(nums.begin(), nums.end());
        while(r>l){
            int val=nums[l]+nums[r];
            if(val==target){
                break;
            }else{
                if(val<target)l++;
                else r--;
            }
        }
        vector<int>ans(2);
        bool flag=0;
        for(int i=0;i<nums.size();i++){
            if(arr[i]==nums[l]&&!flag){ans[0]=i;flag=1;}
            else if(arr[i]==nums[r])ans[1]=i;
        }
        return ans;
    }
};